### PR TITLE
[ADD] product_variant_specific_tax

### PR DESCRIPTION
**Pull request by p-tombez**
Add module which allow to specify additional taxes per variant (e.g. variant with/without battery will not have the same taxes)

### CODE REVIEW

**Review by grindtildeath — COMMENTED** [review]
Looks ok, but would be great to have a basic test applying the additional tax on sale.order.line to avoid regressions ;)

**Review by Tonow-c2c — APPROVED** [review]

**Comment by gurneyalex** [comment]
@p-tombez I just restarted the build on the 11.0 branch which was green but 2 months old, but it looks like this module could have broken the branch. Can you check this?

File: product_variant_specific_tax/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    _inherit = 'sale.order.line'
+
+    def _compute_tax_id(self):
+        super()._compute_tax_id()
+        for line in self:
+            line.tax_id = [
+                (
+                    6,
+                    0,
+                    line.tax_id.ids + line.product_id.additional_tax_ids.ids,
```
Comment (gurneyalex) [inline product_variant_specific_tax/models/sale_order_line.py]: Any special reason not to use `line.tax_id |= line.product_id.additional_tax_ids`. For perf reasons, I'd recommend not writing on lines where `line.product_id.additional_tax_ids` is not set.

File: product_variant_specific_tax/views/product_view.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" ?>
+<odoo>
+
+    <record model="ir.ui.view" id="product_normal_form_view">
+        <field name="name">product_normal_form_view</field>
+        <field name="model">product.product</field>
+        <field name="inherit_id" ref="product.product_normal_form_view" />
+        <field name="arch" type="xml">
+            <field name="taxes_id" position="after">
+                <field name="additional_tax_ids" widget="many2many_tags" />
```
Comment (gurneyalex) [inline product_variant_specific_tax/views/product_view.xml:10]: Does this show up also when you display the variants of a product.template?

**Comment by p-tombez** [comment]
@gurneyalex Something's wrong inside `sale_variant_configurator` addon.
I tested this way : 
1. Install `sale_variant_configurator` (+ dependencies)
2. Run tests in `sale_variant_configurator` => success
3. Install `stock`
4. Run again tests in `sale_variant_configurator` => failed

IMO, nothing to do with the implementation in this PR

**Comment by pedrobaeza** [comment]
@p-tombez There's a module `sale_stock_product_configurator` that acts as glue module and avoids problems when `stock` is installed.

**Comment by p-tombez** [comment]
@pedrobaeza I installed it and redo the tests in `sale_variant_configurator` but they still fails

**Comment by pedrobaeza** [comment]
I can't say then...

Is it red in main branch? If no, then something on this module is altering the result of the other test.

**Comment by p-tombez** [comment]
I ran all the tests without installing the module from this PR.
The only thing that differs between the tests I ran and how Travis run them is that I installed `stock` module

**Comment by gurneyalex** [comment]
@p-tombez @pedrobaeza Branch 11.0 is red on travis https://travis-ci.org/OCA/product-variant/jobs/614220875#L1180 with the same error as this PR -> The red travis is unrelated to the PR.

**Review by gurneyalex — APPROVED** [review]

**Comment by pedrobaeza** [comment]
OK, you can go ahead then

**Pull request merged**